### PR TITLE
support Qnn 2 28

### PR DESCRIPTION
**Pull request by HectorSVC**
### Description
support Qnn 2.28
update default qnn vesion to 2.28 in build pipeline

### CODE REVIEW

File: onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc [hunk 1]
```diff
@@ -641,12 +641,15 @@ Status QnnBackendManager::LoadCachedQnnContextFromBuffer(char* buffer, uint64_t
...
+  if (binary_info->version == QNN_SYSTEM_CONTEXT_BINARY_INFO_VERSION_3) {
+    graph_count = binary_info->contextBinaryInfoV3.numGraphs;
+    graphs_info = binary_info->contextBinaryInfoV3.graphs;
   } else if (binary_info->version == QNN_SYSTEM_CONTEXT_BINARY_INFO_VERSION_2) {
     graph_count = binary_info->contextBinaryInfoV2.numGraphs;
     graphs_info = binary_info->contextBinaryInfoV2.graphs;
+  } else if (binary_info->version == QNN_SYSTEM_CONTEXT_BINARY_INFO_VERSION_1) {
+    graph_count = binary_info->contextBinaryInfoV1.numGraphs;
+    graphs_info = binary_info->contextBinaryInfoV1.graphs;
   }
```
Comment (jywu-msft) [inline onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc:653]: do we need an else case for unknown/unsupported version? (we should have had one before to gracefully handle V3 being unsupported)

File: onnxruntime/test/providers/qnn/gather_op_htp_test.cc [hunk 2]
```diff
@@ -131,11 +131,15 @@ TEST_F(QnnHTPBackendTests, GatherOp_IndicesDynamicInt32_Axis0) {
                                        ExpectedEPNodeAssignment::All);
 }
 
+// disabled for QNN 2.28.0.241029 failed for accuracy validation
```
Comment (jywu-msft) [inline onnxruntime/test/providers/qnn/gather_op_htp_test.cc:134]: we should report these new failures

Comment (HectorSVC) [reply onnxruntime/test/providers/qnn/gather_op_htp_test.cc:134]: yes

File: onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc (hunk 1 above)
Comment (HectorSVC) [reply onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc:655]: updated

**Review by adrianlizarraga — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/android-arm64-v8a-QNN-crosscompile-ci-pipeline.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ parameters:
 - name: QnnSdk
   displayName: QNN SDK version
   type: string
-  default: 2.27.0.240926
+  default: 2.28.0.241029
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/android-arm64-v8a-QNN-crosscompile-ci-pipeline.yml:35]: FYI about https://github.com/microsoft/onnxruntime/pull/22434 which is adding another place where the version is specified.

**Review by jywu-msft — APPROVED** [review]

**Comment by snnn** [comment]
This PR has been cherry-picked into the `rel-1.20.1` branch in PR #22785. Removing the `release:1.20.1` label.

**Pull request merged**